syntax = "proto3";

import "google/protobuf/empty.proto";

option java_multiple_files = true;
option java_package = "io.grpc.cache.plugin.service";
option java_outer_classname = "PluginProto";

package pluginproto;

service PluginProto {

  rpc GetHint(FileHint) returns (FileHint) {}
  rpc UpdateStats(FileRequest) returns (google.protobuf.Empty) {}
  rpc ResetHistory(google.protobuf.Empty) returns (google.protobuf.Empty) {}

}

message FileRequest {
  string filename = 1;
  float downloaded = 2;
  bool hit = 3;
}

message FileHint {
  string filename = 1;
  bool store = 2;
}
